### PR TITLE
Homework8

### PR DESCRIPTION
**Pull request by Ihor-Kaz**
added homework8
Please, check it

### CODE REVIEW

File: homework8_max_3.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# merge the original list and the converted one
+merged = l1 + converted
+
+# sort list
+sorted_list = sorted(merged, reverse=True)
+
+# take 3 max int and multiplication them
+result = sorted_list[0] * sorted_list[1] * sorted_list[2]
+
+print(f'Max value = "{result}". Nums are: {sorted_list[:3]}')
```
Comment (antohaUa) [inline homework8_max_3.py]: From the very beginning incorrect answer 
Your: Max value = "2016". Nums are: [24, 12, 7]
Expected: Max value = "2016". Nums are: (-7, 12, -24)

**Review by antohaUa — CHANGES_REQUESTED** [review]

File: homework8_max_3.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+"""
+
+l1 = [-7, 1, 2, 0, 3, 12, -24, 5, -1, 6]
+
+# create a list where each element is squared
+converted = [num**2 for num in l1]
+
+# sort list
+sorted_list = sorted(converted, reverse=True)
```
Comment (antohaUa) [inline homework8_max_3.py]: Please check your solution vs 
l1 = [-7, 0, 1, 16, 24]
Max value_l1 = 384. Nums are: (24,16,1)

In you case 
Max value = "-2688". Nums are: [24, 16, -7]
384 > -2688   Maximum should be 384

**Review by antohaUa — CHANGES_REQUESTED** [review]

File: homework8_max_3.py (hunk 2 above)
Comment (Ihor-Kaz) [reply homework8_max_3.py]: reworked in other way

File: homework8_max_3.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    Input: [-7, 1, 2, 0, 3, 12, -24, 5, -1, 6]
+    Output: Max value = "2016". Nums are: (-7, 12, -24)
+
+"""
+
+l1 = [-7, 1, 2, 0, 3, 12, -24, 5, -1, 6]
+
+# create a dict where key = multiplication and value = (a, b, c)
+d = {}
+for a in range(len(l1)):
```
Comment (antohaUa) [inline homework8_max_3.py:38]: Correct solution. But complexity = O(n^3)
Such solution acceptable for junior level only.

Please pay attention to sorted list [-24, -7, -1, 0, 1, 2, 3, 5, 6, 12] here possible options to get max are 
el[0] * el[1]* el[-1]  or el[-1] * el[-2]* el[-3]   so you need just to compare which group gives max. And that's it )
Complexity = (O(1) + O(1) +O(1) )*2 = O(1)

**Review by antohaUa — APPROVED** [review]

**Pull request merged**